### PR TITLE
[common-library] fix truncate to dns corner case

### LINKED ISSUE #902: [common-library] Tests for `newrelic.common.naming.truncateToDNSWithSuffix` are incorrect
### Bug description

The tests are located here: https://github.com/newrelic/helm-charts/blob/a0b1859/library/CHART-TEMPLATE/tests/cl-naming_suffix_test.yaml#L16-L22

Tests are passing but they are incorrect:

`% echo -n this-job-suffix-is-too-long-so-it-is-not-possible-to-include-at | wc -c`
> 63

`% echo -n including-full-name-as-it-is-would-way-be-longer-than-job-suffix | wc -c`
> 64

### What happened?

`helm install newrelic/nri-bundle new-relic-agent -f values.yaml`

With a proper values file that has a cluster and a license key, it should template but fails with this error:

> `Error: failed pre-install: warning: Hook pre-install nri-bundle/charts/nri-metadata-injection/templates/admission-webhooks/job-patch/job-createSecret.yaml failed: Job.batch "new-relic-agent-installer-nri-metadata-injectio-admission-create" is invalid: spec.template.labels: Invalid value: "new-relic-agent-installer-nri-metadata-injectio-admission-create": must be no more than 63 characters`

### What you expected to happen?**

The name should be templated correctly so the Helm releases could install properly.

### Anything else we need to know?

A workaround until we solve this issue is to make the release name shorter than 25 chars.

### PR DESCRIPTION
**Pull request by sigilioso**
#### Is this a new chart
No

#### What this PR does / why we need it:
It fixes a bug in the common.library's  `truncateToDNSWithSuffix` helper.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #902 

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [ ] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[mychartname]`)

### CODE REVIEW

**Comment by sigilioso** [comment]
We would need to merge #907 in order to the unrelated fix spellcheck issues.

File: library/CHART-TEMPLATE/tests/cl-naming_suffix_test.yaml [hunk 1]
```diff
@@ -15,17 +15,17 @@ tests:
...
   - it: suffix is longer than maximum length supported
     set:
       fullnameOverride: fno
-      jobSuffix: this-job-suffix-is-too-long-so-it-is-not-possible-to-include-at-it-is
+      jobSuffix: this-job-suffix-is-too-long-so-it-is-not-possible-to-include-as-it-is
     asserts:
       - equal:
           path: metadata.name
-          value: this-job-suffix-is-too-long-so-it-is-not-possible-to-include-at
+          value: this-job-suffix-is-too-long-so-it-is-not-possible-to-include-as
```
Comment (kang-makes) [inline library/CHART-TEMPLATE/tests/cl-naming_suffix_test.yaml:31]: NIT: Could we add tests for prefixes and suffixes with exactly 62 and 63 chars?

My rationale is to avoid other issues that we could have in case we fail again doing the maths. I would also want to test the behavior of the dash in these extreme cases.

**Review by kang-makes — APPROVED** [review]
`% echo -en "including-full-name-as-it-is-would-be-way-longer-tha-job-suffix" | wc -c`
```
63
```

`% echo -en this-job-suffix-is-too-long-so-it-is-not-possible-to-include-as | wc -c`
```
63
```

Left a nit but LGTM.

File: library/CHART-TEMPLATE/tests/cl-naming_suffix_test.yaml (hunk 1 above)
Comment (sigilioso) [reply library/CHART-TEMPLATE/tests/cl-naming_suffix_test.yaml:31]: Sure, thanks for pointing out!

Comment (sigilioso) [reply library/CHART-TEMPLATE/tests/cl-naming_suffix_test.yaml:31]: I added some tests to cover those cases here: [9926128](https://github.com/newrelic/helm-charts/pull/910/commits/9926128253f4ba1396fe2013284acf80ca0c5432)
